File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/network/service/PaymentService.java
package hottopic.mit.co.nz.cleaningservice.network.service;

import hottopic.mit.co.nz.cleaningservice.entities.network.response.BooleanResponse;
import hottopic.mit.co.nz.cleaningservice.entities.network.response.DiscountsResponse;
import hottopic.mit.co.nz.cleaningservice.entities.network.params.PaymentParams;
import hottopic.mit.co.nz.cleaningservice.entities.network.response.UserInfoResponse;
import io.reactivex.Single;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.PUT;

public interface PaymentService {

    @GET("cleaning/payment/discount")
    Single<DiscountsResponse> getDiscouts();

    @PUT("cleaning/payment/balance")
    Single<UserInfoResponse> paymentByBalance(@Body PaymentParams paymentBalance);

    @PUT("cleaning/payment/card")
    Single<BooleanResponse> paymentByCard(@Body PaymentParams paymentBalance);
}

File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/network/service/PaymentService.kt
package hottopic.mit.co.nz.cleaningservice.network.service

import hottopic.mit.co.nz.cleaningservice.entities.network.response.BooleanResponse
import hottopic.mit.co.nz.cleaningservice.entities.network.response.DiscountsResponse
import hottopic.mit.co.nz.cleaningservice.entities.network.params.PaymentParams
import hottopic.mit.co.nz.cleaningservice.entities.network.response.UserInfoResponse
import io.reactivex.Single
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PUT

interface PaymentService {

    @get:GET("cleaning/payment/discount")
    val discouts: Single<DiscountsResponse?>?

    @PUT("cleaning/payment/balance")
    fun paymentByBalance(@Body paymentBalance: PaymentParams): Single<UserInfoResponse?>?

    @PUT("cleaning/payment/card")
    fun paymentByCard(@Body paymentBalance: PaymentParams): Single<BooleanResponse?>?
}
